app/deepseg.cc

```diff
@@ -256,6 +263,57 @@ static bool is_number(const std::string &s) {
 	return !s.empty() && std::all_of(s.begin(), s.end(), ::isdigit);
 }
 
+std::string resolve_path(const std::string& provided, const std::string& type) {
+	std::string result;
+	// Check for network (URI) schema and return as-is
+	// https://www.rfc-editor.org/rfc/rfc3986#section-3.1
+	// however we require at least two chars in the scheme to allow driver letters to work on Windows..
+	if (std::regex_match(provided, std::regex("^[[:alpha:]][[:alnum:]+-.]{1,}:.*$")))
+		return provided;
+	// We use std::ifstream to check we can open each test path read-only, in order:
+	// 1. exactly what was provided
+	if (std::ifstream(provided).good())
+		return provided;
```

Comment:
As yet - this doesn't stop if the `provided` path contains a '/' (as suggested by @BenBE in #101), so it will look for sub-trees in other locations.. useful or confusing?

Reply:
The idea behind skipping the search for sub-trees was to a) mimic the behavior when looking for programs in the PATH and b) to allow for relative paths to always work as expected.

Reply:
OK, emulating PATH search makes intuitive sense, I'm now wondering if we should support a `BACKSCRUB_PATH` ?

Reply:
If it's not too much complexity code-wise go for it.